dev/tools/deploy-to-kube

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+# Copyright 2025 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import os
+import subprocess
+import sys
+import io
+
+import yaml
+
+# Add shared directory to Python path
+sys.path.append(os.path.join(os.path.dirname(__file__), "shared"))
+import utils
+
+def find_and_replace_images(doc, service_name, image_prefix, image_tag):
+    """Recursively finds and replaces container images in a Kubernetes manifest."""
+    def fn(field_path, v):
+        if field_path == ".spec.template.spec.containers[].image":
```

Comment:
Does this work after https://github.com/kubernetes-sigs/agent-sandbox/pull/27? If so, perhaps we should stick with template for pod templates.

Reply:
So this one is replacing the image in the deployment/statefulset.  There's no way in general to discover image paths based on the schema (that I know of?), so I think what everyone does is to hard-code a list of well-known paths.  I guess that's an argument to make it spec.template for consistency, but it's not a huge deal to have another path here (and there are some missing e.g. initContainers)

The one exception is ko, which invents an invalid image syntax `ko://` and thus we know that this is an image.  I'd rather not do that because I don't think we should be taking a dependency on ko right now...